src/model/consent/consent.ts

```diff
@@ -42,45 +42,62 @@
  * of 'createdAt' field may need to be tested in the future.
  */
 
-import { NotFoundError, RevokedConsentModificationError } from '../errors'
-import Knex from 'knex'
+import { NotFoundError } from '../errors'
+import Knex from 'knex';
 
 /*
  * Interface for Consent resource type
  */
 export interface Consent {
   id: string;
-  initiatorId?: string;
-  participantId?: string;
+  // participant DFSP that requested a Consent resource be made
+  participantId: string;
+  // status of the consent
   status: string;
-  credentialId?: string;
-  credentialType?: string;
-  credentialStatus?: string;
-  credentialPayload?: string;
-  credentialChallenge?: string;
+  // NOTE: not sure what purpose credentialId serves.
+  credentialId: string;
+  // credential type - currently trying to support FIDO/Generic credentials
+  credentialType: 'FIDO' | 'GENERIC' | undefined;
+  // NOTE: unsure this field is needed since `auth-service` verifies
+  //       credential on receiving them
+  credentialStatus: string;
+  // assuming this is the public key of the pair
+  credentialPayload: string;
```

Comment:
I think this should be type `string` | `FIDOPublicKeyCredentialAttestation`, depending on the `credentialType`

Reply:
`Consent` interface is 1:1 with the db are you suggesting we store all the fields from `FIDOPublicKeyCredentialAttestation`?

Looking at https://www.npmjs.com/package/fido2-lib I'm just assuming we need to store the public key from `await f2l.attestationResult` for FIDO at least

Reply:
I saw some old code that pointed me to the assumption that `credentialPayload` is a public key. Can't find it atm...

Reply:
Yeah, you're right. The `f2l.attestationResult` should do the trick